examples/cms/src/App.tsx

```diff
@@ -0,0 +1,86 @@
+import "./App.css"
+
+import { framer, ManagedCollection } from "framer-plugin"
+import { useEffect, useLayoutEffect, useState, useRef } from "react"
+import { DataSource, getDataSource, getDataSourcesIds } from "./data"
+import { FieldMapping } from "./FieldMapping"
+import { SelectDataSource } from "./SelectDataSource"
+import { Spinner } from "./components/Spinner"
+
+interface AppProps {
+    collection: ManagedCollection
+    previouslyConfiguredDataSourceId: string | null
+    previouslyConfiguredSlugFieldId: string | null
+}
+
+export function App({ collection, previouslyConfiguredDataSourceId, previouslyConfiguredSlugFieldId }: AppProps) {
+    const [dataSource, setDataSource] = useState<DataSource | null>(null)
+
+    const hasStartedLoadingDataSource = useRef(false)
+    const [isLoadingConfiguredDataSource, setIsLoadingConfiguredDataSource] = useState(
+        Boolean(previouslyConfiguredDataSourceId)
+    )
+    const hasDataSourceSelected = Boolean(isLoadingConfiguredDataSource || dataSource)
+
+    useLayoutEffect(() => {
+        framer.showUI({
+            width: hasDataSourceSelected ? 360 : 320,
+            height: hasDataSourceSelected ? 425 : 305,
+            minWidth: hasDataSourceSelected ? 360 : undefined,
+            minHeight: hasDataSourceSelected ? 425 : undefined,
+            resizable: dataSource !== null,
```

Comment:
Any reason we use `hasDataSourceSelected` for all layout states except for `resizable`?

Reply:
Yes, plugins remember their previous window dimensions, so we need to be careful about when we allow resizing to prevent ending up with an oversized window showing just a small spinner!

`dataSource !== null` is specifically used for resizable because we only want resizing enabled when we actually have the data source loaded, not during the loading state